caffe2/core/tensor_impl.h

```diff
@@ -88,6 +88,7 @@ inline int canonical_axis_index_(int axis_index, int ndims) {
 class CAFFE2_API TensorImpl : public c10::intrusive_ptr_target {
  public:
   TensorImpl() = delete;
+  explicit TensorImpl(at::Storage&& storage) : storage_(storage) {}
```

Comment:
Can you pass in Storage by value instead?

Reply:
also, even when passing it in as `&&`, you might have to repeat `std::move()` when assigning it to the member. I think rvalue references decay.